CMakeLists.txt

```diff
@@ -409,9 +409,6 @@ endif()
 # Native TLS configuration
 ################################################################################
 set(HPX_WITH_NATIVE_TLS_DEFAULT ON)
-if(APPLE)
-  set(HPX_WITH_NATIVE_TLS_DEFAULT ${HPX_WITH_CXX11_THREAD_LOCAL})
-endif()
```

Comment:
This means that mac users need Xcode >= 8, but I think that's a reasonable requirement as it was released in 2016. One less thing to worry about.